pkg/ebpf/perf/event.go

```diff
@@ -0,0 +1,213 @@
+	}
+}
+
+// Flush flushes the pending data from the underlying perfbuf/ringbuf
+func (e *EventHandler) Flush() {
+	e.f.Flush()
+}
+
+// ResizeRingBuffer resizes the ring buffer by creating/updating a map spec editor
+func ResizeRingBuffer(mgrOpts *manager.Options, mapName string, bufferSize int) {
+	if mgrOpts.MapSpecEditors == nil {
+		mgrOpts.MapSpecEditors = make(map[string]manager.MapSpecEditor)
+	}
+	specEditor := mgrOpts.MapSpecEditors[mapName]
+	specEditor.MaxEntries = uint32(bufferSize)
+	specEditor.EditorFlag |= manager.EditMaxEntries
+	mgrOpts.MapSpecEditors[mapName] = specEditor
+}
+
+func (e *EventHandler) initPerfBuffer(mgr *manager.Manager) {
+	mgr.PerfMaps = slices.DeleteFunc(mgr.PerfMaps, func(perfMap *manager.PerfMap) bool {
+		return perfMap.Name == e.opts.MapName
+	})
+	pm := &manager.PerfMap{
+		Map: manager.Map{Name: e.opts.MapName},
+		PerfMapOptions: manager.PerfMapOptions{
+			PerfRingBufferSize: e.opts.PerfOptions.BufferSize,
+			Watermark:          e.opts.PerfOptions.Watermark,
+			WakeupEvents:       e.opts.PerfOptions.WakeupEvents,
+			RecordHandler:      e.perfRecordHandler,
+			LostHandler:        nil, // TODO do we need support for Lost?
+			RecordGetter:       perfPool.Get,
+			TelemetryEnabled:   e.opts.TelemetryEnabled,
+		},
+	}
+	mgr.PerfMaps = append(mgr.PerfMaps, pm)
+	ebpfTelemetry.ReportPerfMapTelemetry(pm)
+	e.f = pm
+}
+
+func (e *EventHandler) perfRecordHandler(record *perf.Record, _ *manager.PerfMap, _ *manager.Manager) {
+	defer perfPool.Put(record)
+	e.opts.Handler(record.RawSample)
+}
+
+func (e *EventHandler) initRingBuffer(mgr *manager.Manager) {
+	mgr.RingBuffers = slices.DeleteFunc(mgr.RingBuffers, func(ringBuf *manager.RingBuffer) bool {
+		return ringBuf.Name == e.opts.MapName
+	})
+	rb := &manager.RingBuffer{
+		Map: manager.Map{Name: e.opts.MapName},
+		RingBufferOptions: manager.RingBufferOptions{
+			RecordHandler:    e.ringRecordHandler,
+			RecordGetter:     ringbufPool.Get,
+			TelemetryEnabled: e.opts.TelemetryEnabled,
+		},
+	}
+	mgr.RingBuffers = append(mgr.RingBuffers, rb)
+	ebpfTelemetry.ReportRingBufferTelemetry(rb)
+	e.f = rb
+}
+
+func (e *EventHandler) ringRecordHandler(record *ringbuf.Record, _ *manager.RingBuffer, _ *manager.Manager) {
+	defer ringbufPool.Put(record)
+	e.opts.Handler(record.RawSample)
+}
+
+// UpgradePerfBuffer upgrades a perf buffer to a ring buffer by creating a map spec editor
+func UpgradePerfBuffer(mgr *manager.Manager, mgrOpts *manager.Options, mapName string) {
+	if mgrOpts.MapSpecEditors == nil {
+		mgrOpts.MapSpecEditors = make(map[string]manager.MapSpecEditor)
+	}
+	specEditor := mgrOpts.MapSpecEditors[mapName]
+	specEditor.Type = ebpf.RingBuf
+	specEditor.KeySize = 0
+	specEditor.ValueSize = 0
+	specEditor.EditorFlag |= manager.EditType | manager.EditKeyValue
+	mgrOpts.MapSpecEditors[mapName] = specEditor
+
+	mgr.PerfMaps = slices.DeleteFunc(mgr.PerfMaps, func(perfMap *manager.PerfMap) bool {
+		return perfMap.Name == mapName
+	})
```

Comment:
Why do we have that here? It is already exist in `initRingBuffer` which will called if `UpgradePerfBuffer` was called